nym-network-monitor/src/accounting.rs

```diff
@@ -384,17 +424,73 @@ async fn submit_gateway_stats_to_db(client: Arc<Client>) -> anyhow::Result<()> {
     Ok(())
 }
 
-pub async fn submit_metrics_to_db(client: Arc<Client>) -> anyhow::Result<()> {
+async fn db_connection(database_url: Option<&String>) -> Result<Option<(Client, JoinHandle<()>)>> {
+    if let Some(database_url) = database_url {
+        let (client, connection) = tokio_postgres::connect(database_url, NoTls).await?;
+
+        let handle = tokio::spawn(async move {
+            if let Err(e) = connection.await {
+                error!("Postgres connection error: {}", e);
+            }
+        });
+
+        Ok(Some((client, handle)))
+    } else {
+        Ok(None)
+    }
+}
+pub async fn submit_metrics_to_db(database_url: Option<&String>) -> anyhow::Result<()> {
+    if let Some((client, handle)) = db_connection(database_url).await? {
+        let client = Arc::new(client);
+        let client2 = Arc::clone(&client);
+        let client3 = Arc::clone(&client);
```

Comment:
why are you cloning all those pointers? I'm not sure you have to (unless you wanted to execute all of those submissions in parallel)

Reply:
Since all of them take an Arc its possible I wanted to execute them in parallel, maybe DB was complaining, interesting

Reply:
Arc is implicitly moved always, hence the cloning, ie just passing the same Arc will not work